tests-foundry/unit/BurnTest.t.sol

```diff
@@ -0,0 +1,139 @@
+        burnCap = _limitFunds(burnCap); 
+
+        _updateBurnCap(burnCap);        
+
+        vm.deal(address(s_burn), addedBalance);
+        s_burn.burn{value: value}();
+    }
+
+    function testFuzz_burn_with_burnCap_lesser_than_balance_results_in_no_eth_transfer(uint burnCap) public {
+        burnCap = _limitFunds(burnCap);
+
+        if (address(s_burn).balance <= burnCap) {
+            vm.deal(address(s_burn), burnCap+1); 
+        }
+        
+        _updateBurnCap(burnCap); 
+
+        assertTrue( burnCap >= address(s_burn).balance, "bad burnCap value");
+
+        address sender = makeAddr("Joe");
+        
+        uint value = burnCap; 
+        _hoaxWithGas(sender, value); // internally adds ADDITIONAL_GAS_FEES to value
+
+        uint preSenderBalance = address(sender).balance;
+
+        s_burn.burn{value: value}();
+
+        uint postSenderBalance = address(sender).balance;
+
+        console.log("preSenderBalance: %s, postSenderBalance: %s", preSenderBalance, postSenderBalance);
+
+        assertTrue(postSenderBalance > preSenderBalance - ADDITIONAL_GAS_FEES , "no eth should have been transferred to sender");
+    }
+
+
+    // --- updateParam() tests ---
+
+    function test_verify_nonGov_cannot_updateParams() public {
+        uint legalBurnCapValue = address(s_burn).balance + 1;
+        vm.prank(makeAddr("Joe"));
+        vm.expectRevert(abi.encodePacked(ONLY_GOV_ALLOWED_MSG));
+        s_burn.updateParam(BURN_CAP_KEY, abi.encodePacked(legalBurnCapValue));
+    }
+
+    function testFuzz_updateParams_success_scenario(uint newBurnCap,uint addedBalance) public {
+        vm.deal(address(s_burn), addedBalance);
+        vm.assume( newBurnCap >= address(s_burn).balance); //@test?? 
+
+        _updateBurnCap(newBurnCap);        
+
+        assertEq(newBurnCap, s_burn.burnCap(), "bad cap value");
+    }
+
+    function testFuzz_updateParams_bad_paramName(string memory paramName, uint addedBalance) public {
+        vm.assume(!_strEqual(paramName, BURN_CAP_KEY)); // else no BAD_PARAM_ERROR_MSG revert
+        addedBalance = _limitFunds(addedBalance); 
+        uint origBurnCap = s_burn.burnCap();
+        vm.deal(address(s_burn), addedBalance);                
+        vm.prank(GOV_HUB_ADDR);
+
+        uint legalBurnCapValue = address(s_burn).balance + 1;
+
+        vm.expectRevert(abi.encodePacked(BAD_PARAM_ERROR_MSG));
+        s_burn.updateParam(paramName, abi.encodePacked(legalBurnCapValue));
+
+        assertEq(origBurnCap, s_burn.burnCap(), "bad cap value");
+    }
+
+    function testFuzz_updateParams_with_bad_burnBap(uint newBurnCap, uint addedBalance) public {
+        addedBalance = _limitFunds(addedBalance); 
+        vm.deal(address(s_burn), addedBalance);
+
+        uint origBurnCap = s_burn.burnCap();
+        uint origBalance = address(s_burn).balance;
+
+        vm.assume( newBurnCap < address(s_burn).balance); // should result in OutOfBounds error
+        
+        vm.expectRevert(
```

Comment:
with result in test failure unless the next call reverts with an 'OutOfBounds' error